icloud.com.custom_email_domain.json

```diff
@@ -0,0 +1,50 @@
+{
+   "providerId":"icloud.com",
+   "providerName":"iCloud",
+   "serviceId":"custom_email_domain",
+   "serviceName":"Custom Email Domain",
+   "version": 1,
+   "logoUrl":"",
+   "description":"This is a template to enable iCloud Custom Email Domain",
+   "syncPubKeyDomain": "icloudmailadmin.com",
+   "multiInstance": true,
+   "records":[
+      {
+         "groupId":"icloud-verification",
+         "type":"TXT",
+         "host":"@",
+         "data":"%icloud_token%",
+         "ttl":3600
+      },
+      {
+         "groupId":"icloud-spf",
+         "type":"TXT",
```

Comment:
SPF records shall normally use SPFM record type.
Usage of TXT record shall be an exception and avoided as far as possible.

Reply:
@rohitkandhari thanks for correcting other issues.
How about this last one with the use of SPFM?

I understand the convenience of using `redirect` SPF directive, but still it is not compatible with other usages of a domain, which can be well configured in parallel to iCloud. Especially if the domain hosts a website it is essential that the CMS is able to send out emails so the rules have to be merged and not exclusive to iCloud.
Details in the [specification](https://github.com/Domain-Connect/spec/blob/master/Domain%20Connect%20Spec%20Draft.adoc#multiple-services).

A pure TXT record as proposed here may also put a second SPF record if one exists already, rendering invalid SPF configuration.

Also any verification schema which checks exactly the content of SPF must not be employed, because SPF can be changed any time to accomplish desired customer configuration.

Reply:
@rohitkandhari I appreciate a lot your update to SPFM. 

One operational remark you may want to consider would be to use different variable, like `%spfm_data%` in the v2 template, so that you can serve both providers still using v1 template and the ones using v2 template same time, by sending `spf_data` and `spfm_data` variables in parallel for the transition period until all providers switch to v2 template.

I can also approve and merge this template as it is if you confirm you would not consider my suggestion.

Reply:
So what I have done today is check the template version which is returned back from the supported call https://github.com/Domain-Connect/spec/blob/master/Domain%20Connect%20Spec%20Draft.adoc#query-supported-template and then publish the data accordingly.  Is this good or there can be providers who do not send the version back and we can send both the spf_data and spfm_data and the provider would substitute the appropriate value depending upon the template supported?

Reply:
Not all providers reports back template version AFAIK but if your test shows you get the version back from all providers who deployed the template then indeed you are fine as well. Please let me know  if to merge as is.

Reply:
I am going to make few changes to incorporate what we discussed above. Will update one ready so that we can merge the PR.

Reply:
hey @pawel-kow we are good to merge this, thanks for all the good feedback